include/sparrow_ipc/chunk_memory_serializer.hpp

```diff
@@ -130,6 +130,7 @@ namespace sparrow_ipc
         chunked_memory_output_stream<std::vector<std::vector<uint8_t>>>* m_pstream;
         bool m_ended{false};
         std::optional<CompressionType> m_compression;
+        compression_cache_t m_cache;
```

Comment:
I don't understand why this cache is here.
We have to manage the compression in a serialize_record_batch, not in the stream serializer.